src/viewing-keys.ts

```diff
@@ -41,6 +41,7 @@ export class ViewingKeyManager {
       account = this.addAccount();
     }
     if (!account) throw new Error('No account available');
+    if (!key) throw new Error('Empty or undefined key cannot be added');
```

Comment:
Should be this the first thing that you validate in this function?

Reply:
My bad, moving this validation as firts line of the function

Reply:
Thanks